server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/LanguagePluginParser.java

```diff
@@ -0,0 +1,63 @@
+package org.jetbrains.bsp.bazel.server.sync.languages;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LanguagePluginParser {
+  final Pattern PACKAGE_PATTERN = Pattern.compile("^\\s*package\\s+([\\w\\.]+)");
+
+  public LanguagePluginParser() {}
+
+  public String findPackage(Path source, boolean multipleLines) {
+    List<String> packages;
+    File sourceFile = new File(String.valueOf(source));
+    try {
+      BufferedReader reader = new BufferedReader(new FileReader(sourceFile));
+
+      if (multipleLines) {
+        packages = this.readMultipleLines(reader);
+      } else {
+        packages = this.readSingleLines(reader);
+      }
+
+      reader.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+    if (packages != null && !packages.isEmpty()) {
+      return String.join(".", packages);
+    }
+    return null;
+  }
+
+  private List<String> readMultipleLines(BufferedReader reader) throws IOException {
+    List<String> packages = new ArrayList<>();
+    String line;
+    while ((line = reader.readLine()) != null) {
+      Matcher matcher = PACKAGE_PATTERN.matcher(line);
+      if (matcher.find()) {
```

Comment:
the thing i meant was that this method and `readSingleLines` could be merged, and there would be jsut an if statement inside to decide should the result be returned (false) or appended to the list(true)

Reply:
Merged methods